### PR TITLE
docs: replace method links with intra-links

### PR DESCRIPTION
**Pull request by xliiv**
## Motivation
Make doc's links more resistant.

## Solution
Delagates doc's links resolution to [`rustdoc`](https://github.com/rust-lang/rfcs/blob/master/text/1946-intra-rustdoc-links.md).
 
Refs: #1473

### CODE REVIEW

**Comment by xliiv** [comment]
Blocked by https://github.com/rust-lang/rust/pull/71289

File: tokio/src/io/registration.rs [hunk 1]
```diff
@@ -34,9 +34,9 @@ cfg_io_driver! {
     /// stream. The write readiness event stream is only for `Ready::writable()`
     /// events.
     ///
-    /// [`new`]: #method.new
-    /// [`poll_read_ready`]: #method.poll_read_ready`]
-    /// [`poll_write_ready`]: #method.poll_write_ready`]
+    /// [`new`]: method@Self::new
+    /// [`poll_read_ready`]: method@Self::poll_read_ready`]
+    /// [`poll_write_ready`]: method@Self::poll_write_ready`]
```
Comment (Darksonn) [inline tokio/src/io/registration.rs:39]: This seems like it has an extra \`] at the end.

Comment (xliiv) [reply tokio/src/io/registration.rs:39]: Good catch, i'll fix it, thanks!

File: tokio/src/net/udp/split.rs [hunk 2]
```diff
@@ -23,14 +23,14 @@ use std::sync::Arc;
 
 /// The send half after [`split`](super::UdpSocket::split).
 ///
-/// Use [`send_to`](#method.send_to) or [`send`](#method.send) to send
+/// Use [`send_to`](method@Self::send_to) or [`send`](#method.send) to send
```
Comment (taiki-e) [inline tokio/src/net/udp/split.rs:26]:
```suggestion
/// Use [`send_to`](method@Self::send_to) or [`send`](method@Self::send) to send
```

File: tokio/src/net/udp/split.rs [hunk 3]
```diff
@@ -23,14 +23,14 @@ use std::sync::Arc;
...
 /// datagrams.
 #[derive(Debug)]
 pub struct SendHalf(Arc<UdpSocket>);
 
 /// The recv half after [`split`](super::UdpSocket::split).
 ///
-/// Use [`recv_from`](#method.recv_from) or [`recv`](#method.recv) to receive
+/// Use [`recv_from`](method@Self::recv_from) or [`recv`](#method.recv) to receive
```
Comment (taiki-e) [inline tokio/src/net/udp/split.rs:33]:
```suggestion
/// Use [`recv_from`](method@Self::recv_from) or [`recv`](method@Self::recv) to receive
```

File: tokio/src/io/registration.rs (hunk 1 above)
Comment (xliiv) [reply tokio/src/io/registration.rs]: Fixed, thanks!

File: tokio/src/net/udp/split.rs (hunk 2 above)
Comment (xliiv) [reply tokio/src/net/udp/split.rs]: Fixed, thanks!

File: tokio/src/net/udp/split.rs (hunk 3 above)
Comment (xliiv) [reply tokio/src/net/udp/split.rs]: Fixed, thanks!

**Comment by xliiv** [comment]
It's ready for review @Darksonn 
Take a look at changes related to `register`.
It looks like docs leftovers after API changes? Any suggestions are welcome.

File: tokio/src/io/registration.rs [hunk 4]
```diff
@@ -167,9 +165,8 @@ impl Registration {
     ///
     /// * `Poll::Ready(Err(err))` means that the registration has encountered an
     ///   error. This error either represents a permanent internal error **or**
-    ///   the fact that [`register`] was not called first.
+    ///   the fact that `register` was not called first.
```
Comment (xliiv) [inline tokio/src/io/registration.rs:168]: i'm talking about this..

File: tokio/src/io/registration.rs [hunk 5]
```diff
@@ -227,9 +222,8 @@ impl Registration {
...
```
Comment (xliiv) [inline tokio/src/io/registration.rs:225]: and this

**Comment by xliiv** [comment]
No idea why it fails - I'll check it later
Anyway works for me
```shell
[23:43:37] xliiv@alef /home/xliiv/workspace/tokio (0)
> env RUSTDOCFLAGS="--cfg docsrs" cargo +nightly doc --all-features
    Finished dev [unoptimized + debuginfo] target(s) in 0.09s
[23:43:39] xliiv@alef /home/xliiv/workspace/tokio (0)
> rustc +nightly --version
rustc 1.45.0-nightly (8970e8bcf 2020-05-23)
[23:43:45] xliiv@alef /home/xliiv/workspace/tokio (0)
> git show --summary
commit 03510730bc563091b83ea3aabb75146df04d1b6f (HEAD -> 1473-fix-links, xliiv/1473-fix-links)
Author: Tymoteusz Jankowski <tymoteusz.jankowski@gmail.com>
Date:   Fri May 22 22:07:20 2020 +0200

    docs: replace method links with intra-links


```

**Comment by taiki-e** [comment]
CI failure is due to our CI using old nightly rustc:

https://github.com/tokio-rs/tokio/blob/954f2b73048d91ec3613e5db901aa17fd6f31bef/azure-pipelines.yml#L6

**Comment by xliiv** [comment]
@taiki-e is there an upgrade policy? can i udpate?

**Comment by Darksonn** [comment]
I don't think there's an official policy, but we should probably verify that docs.rs uses a similarly new nightly to compile their docs. If they are, I think it is fine to go ahead an update it.

**Comment by xliiv** [comment]
@Darksonn it should be ok, according to this
> The current version of the Rust compiler in use is rustc 1.45.0-nightly (46e85b432 2020-05-24)

from https://docs.rs/about

**Comment by xliiv** [comment]
Just upgraded, let's see if it works now..

BTW I've made it 2 commits on purpose (one extra for the upgrade). Let me know if it's OK.

File: tokio/src/io/registration.rs (hunk 4 above)
Comment (Darksonn) [reply tokio/src/io/registration.rs:168]: It seems like registration is now done in the constructor. The equivalent error now would be calling it _after_ `deregister` has been called.

**Comment by Darksonn** [comment]
It looks good. I commented on the registration thing, and it appears to be rather easy to fix. Once that's changed, we can merge it.

File: tokio/src/io/registration.rs (hunk 4 above)
Comment (xliiv) [reply tokio/src/io/registration.rs]: Removed all mentions of `register`

File: tokio/src/io/registration.rs (hunk 5 above)
Comment (xliiv) [reply tokio/src/io/registration.rs]: Removed all mentions of `register`

**Comment by xliiv** [comment]
@Darksonn, it seems `rustfmt` got fixed, so I've updated the nightly version to the working one.

see
https://rust-lang-nursery.github.io/rust-toolstate/

**Pull request merged**